internal/dag/builder.go

```diff
@@ -493,6 +484,37 @@ func (b *Builder) computeHTTPLoadBalancers() {
 	}
 }
 
+// mergeConditions merges any two conditions when they are delegated
+func mergeConditions(delegatedCondition, includeCondition *projcontour.Condition) *projcontour.Condition {
```

Comment:
suggestion: the signature of this function, and its name tell us it's working with conditions. I think its probably less to read to just write
```
func mergeConditions(delegate, include *projcontour.Condition) *projcontour.Condition
```